docs/proposals/remote-artifacts.md

```diff
@@ -0,0 +1,293 @@
+the same `BuildSource`.
+
+The following snippet shows how an Artifact (`BuildSource`) will be represented.
+
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
+kind: BuildSource
+metadata:
+  name: license-file
+spec:
+  sources:
+    - type: http
+      url: https://licenses.company.com/customer-id/license.tar
+      http:
+        path: $(workspace)/licenses/license.tar
+```
+
+## Usage
+
+The usage of this feature is based on declaring a slice of `.spec.sources` and later on overwriting
+entries using `BuildRun` resource. For instance:
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
+kind: Build
+metadata:
+  name: example
+spec:
+  sources:
+    - name: license-file
+```
+
+Could have its `license-file` overwritten in a `BuildRun` with:
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
+kind: BuildRun
+metadata:
+  name: license-file
+spec:
+  buildRef:
+    name: example
+  sources:
+    - name: license-file
+      sourceRef: alternative-file
+```
+
+## Steps and Helper
+
+Artifacts are requirements for the build process, thus downloading and preparing artifacts must
+happen before the build process starts. To achieve the objective, the operator will generate a new
+task step to download the artifacts.
+
+The download may happen using existing open-source software, more specifically `wget`. The
+container image which contains this software will be specified via environment variable, and when
+not informed will use a default value.
+
+Therefore, the operator needs to generate `wget` commands with arguments in order to download the
+artifacts specified.
+
+The step generated by the operator will look like the following:
+
+```yml
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: artifact-download
+spec:
+  steps:
+    - name: artifact-download
+      image: busybox:latest
+      command:
+        - /bin/bash
+      args:
+        - -c
+        - >
+          wget --output="/workspace/source/classpath/main.jar" https://nexus.company.com/app/main.jar && \
+          chown 1000:1001 /workspace/source/classpath/main.jar && \
+          chmod 0644 /workspace/source/classpath/main.jar
+```
+
+## Example Use-Case
+
+Using [Shipwright's proposed logos](https://github.com/shipwright-io/build/issues/325) as example,
+let's assume we are building a [TypeScript application](https://github.com/otaviof/typescript-ex)
+which will use the project logo, and we would like to create two different builds, one with the
+default project logo, and another with the alternative.
+
+By using remote artifacts, we can keep the separation of project source code and assets, and we
+can describe those resources as:
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
+kind: BuildSource
+metadata:
+  name: ship-logo
+spec:
+  sources:
+    - name: project-logo
+      type: http
+      url: https://user-images.githubusercontent.com/2587818/92114986-69bfb600-edfa-11ea-820e-96cdb1014f58.png
```

Comment:
Could you use a more realistic example like a jar artifact in all examples, please?

Reply:
Same as in [here](https://github.com/shipwright-io/build/pull/419/files#r520463394).